File: src/main/java/com/pack/colorchat/ChatService.java
package com.pack.colorchat;

import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

@Component
public class ChatService {

    private List<Message> messages = Collections.synchronizedList(new ArrayList<>());

    public void addMessage(User user, String text) {
        messages.add(new Message(user, text));
    }

    public List<Message> getMessages() {
        return messages;
    }
}

File: src/main/kotlin/com/pack/colorchat/service/ChatService.kt
package com.pack.colorchat.service

import com.pack.colorchat.model.Message
import com.pack.colorchat.model.User

const val KOTLIN_CHAT_SERVICE: String = "Kotlin chat service"

interface ChatService {
    fun addMessage(user: User, text: String)
    fun getMessages(): List<Message>
}
